Docs/andrembcosta/pf fracture

andrembcosta opened the pull request:
Opened this new PR to add the documentation of the phase-field classes, as the old PR became too old and merging develop led to lots of conflicts.

Although I added the function and classes descriptions following https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/developerGuide/Contributing/Doxygen.html, when I compile the doxygen docs with make geosx_doxygen, I still dont the stuff I have added. Any ideas of what could be missing?

jhuang2601 commented:
Have you run `make uncrustify_style` to check and update code format?

andrembcosta commented:
> Have you run `make uncrustify_style` to check and update code format?

just did it

CusiniM commented:
> Opened this new PR to add the documentation of the phase-field classes, as the old PR became too old and merging develop led to lots of conflicts.
> 
> Although I added the function and classes descriptions following https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/developerGuide/Contributing/Doxygen.html, when I compile the doxygen docs with make geosx_doxygen, I still dont the stuff I have added. Any ideas of what could be missing?

Which things are you not seeing? I don't think we build the doxygen for all geosx modules.

CusiniM commented:
Actually, everything you have added seems to be in namespaces for which we don't build the doxygen: https://geosx-geosx.readthedocs-hosted.com/en/latest/doxygen_output/html/namespaces.html

andrembcosta commented:
> > Opened this new PR to add the documentation of the phase-field classes, as the old PR became too old and merging develop led to lots of conflicts.
> > Although I added the function and classes descriptions following https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/developerGuide/Contributing/Doxygen.html, when I compile the doxygen docs with make geosx_doxygen, I still dont the stuff I have added. Any ideas of what could be missing?
> 
> Which things are you not seeing? I don't think we build the doxygen for all geosx modules.

I thought I would see something about the PhaseFieldSolver here: 

![image](https://user-images.githubusercontent.com/39500501/192343139-32144d6e-841c-45a2-bc5d-6a5c1a4c7876.png)

andrembcosta commented:
> > 
> 
> I would say that we are not building the doxygen for `physicsSolvers`. I thought we were building it for `constitutive` but it looks like we are not.

All the phase-field fracture classes are either physicsSolvers or constitutive, so, it is probably ok then?

CusiniM commented on src/coreComponents/constitutive/solid/DamageSpectral.hpp:
```diff
@@ -246,15 +288,15 @@ class DamageSpectralUpdates : public DamageUpdates< UPDATE_BASE >
   }
 
-
+  ///accessor for the energy threshold, which, in the cohesive formulation, is independent of the regularization length
```
```suggestion
/**
  * @brief accessor for the energy threshold, which, in the cohesive formulation, is independent of the regularization length
  * return the critical energy threshold value.
  */
```

andrembcosta replied:
is this because the single-line comment is too long?

CusiniM commented on src/coreComponents/constitutive/solid/DamageSpectralUtilities.hpp:
```diff
@@ -33,7 +33,13 @@
+ *@param[out] positivePart Array that stores the positive part of T in Voigt Notation.
+ */
 GEOSX_HOST_DEVICE inline
 void PositivePartOfTensor( real64 (& eigs)[3], real64 (& eigvecs)[3][3], real64 (& positivePart)[6] )
```
can you also make the function names lowercase?

andrembcosta replied:
ok

rrsettgast commented:
@frankfeifan Can you take a look at this and pick it up?
